core/src/main/java/org/apache/accumulo/core/classloader/ClassLoaderUtil.java

```diff
@@ -75,22 +75,29 @@ static synchronized void resetContextFactoryForTests() {
 
   @SuppressWarnings("deprecation")
   public static ClassLoader getClassLoader(String context) {
-    try {
-      if (FACTORY.isValid(context)) {
-        return FACTORY.getClassLoader(context);
-      } else {
-        return org.apache.accumulo.start.classloader.vfs.AccumuloVFSClassLoader.getClassLoader();
-      }
-    } catch (RuntimeException e) {
+    if (context != null && !context.isEmpty()) {
+      return FACTORY.getClassLoader(context);
+    } else {
       return org.apache.accumulo.start.classloader.vfs.AccumuloVFSClassLoader.getClassLoader();
     }
   }
 
   public static boolean isValidContext(String context) {
```

Comment:
`isValidContext` is only called from `PropUtil.validateProperties`. What you have below is very similar to my [original](https://github.com/apache/accumulo/pull/3678/commits/03f0b362463cdb2d17233a7928fd7efee0fbb518#diff-10adcecab49078fcfd1a0694c44f8f67fb0bbcc45bfe89bc624b794de480e3fdR68) commit in #3678. @ctubbsii comments on that are [here](https://github.com/apache/accumulo/pull/3678#pullrequestreview-1563829134). Again, I think we'll need to reconcile the difference of opinion.

Reply:
I'm fine with avoiding the SPI change, with the simpler implementation.